lib/node_modules/@stdlib/math/base/complex/divide/lib/cdiv.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+// MODULES //
+
+var FLOAT64_BIGGEST = require( '@stdlib/math/constants/float64-max' );
+var FLOAT64_SMALLEST = require( '@stdlib/math/constants/float64-smallest-normal' );
+var EPS = require( '@stdlib/math/constants/float64-eps' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var max = require( '@stdlib/math/base/special/max' );
+var isArray = require( '@stdlib/assert/is-array' );
+var robustInternal = require( './robust_internal.js' );
+
+
+// VARIABLES //
+
+var LARGE_THRESHOLD = FLOAT64_BIGGEST * 0.5;
+var SMALL_THRESHOLD = FLOAT64_SMALLEST * ( 2.0 / EPS );
+var RECIP_EPS_SQR = 2.0 / ( EPS * EPS );
+
+
+// MAIN //
+
+/**
+* Divides two complex numbers.
+*
+* ## References
+*
+* * Baudin, Michael, and Robert L. Smith. 2012. "A Robust Complex Division in Scilab" abs/1210.4539 (October). <https://arxiv.org/abs/1210.4539>. https://arxiv.org/pdf/1210.4539.pdf
+*
+* @param {Array<number>} quotient - optional output array
+* @param {number} re1 - real component
+* @param {number} im1 - imaginary component
+* @param {number} re2 - real component
+* @param {number} im2 - imaginary component
+* @returns {Array<number>} real and imaginary components
+*
+* @example
+* var v = cdiv( -13.0, -1.0, -2.0, 1.0 );
+* // returns [ 5.0, 3.0 ]
+*/
+function cdiv( quotient, re1, im1, re2, im2 ) {
+	var ab;
+	var cd;
+	var s;
+
+	if ( arguments.length < 5 ) {
+		im2 = re2;
+		re2 = im1;
+		im1 = re1;
+		re1 = quotient;
+		quotient = [ 0.0, 0.0 ];
+	} else if ( !isArray( quotient ) ) {
```

Comment:
No need to validate output argument. For base implementations, we do not type check (typically). We assume that reasonable input arguments are provided.

Also, reassigning argument variables when using the `arguments` keyword is a no-no for certain JavaScript engines. When doing so, the function is forever deoptimized. So, one option is to not provide a function signature (e.g., `cdiv()`) and extract each argument individually. Or, you can delegate to other functions; e.g.,

``` javascript
cdiv( out, re1, im1, re2, im2 ) {
    if ( arguments.length === 4 ) {
        return divide( new Array( 2 ), out, re1, im1, re2 );
    }
    return divide( out, re1, im1, re2, im2 );
}
```